python/ray/serve/_private/deployment_state.py

```diff
@@ -1595,55 +1579,76 @@ def autoscale(
         current_num_ongoing_requests = self.get_replica_current_ongoing_requests()
         autoscaling_policy = self._target_state.info.autoscaling_policy
         decision_num_replicas = autoscaling_policy.get_decision_num_replicas(
-            curr_target_num_replicas=self._target_state.num_replicas,
+            curr_target_num_replicas=self._target_state.target_num_replicas,
             current_num_ongoing_requests=current_num_ongoing_requests,
             current_handle_queued_queries=current_handle_queued_queries,
+            target_capacity=self._target_state.info.target_capacity,
+            target_capacity_direction=self._target_state.info.target_capacity_direction,
         )
 
-        # Clip the replica count by capacity-adjusted bounds.
-        # TODO (shrekris-anyscale): this should logic should be pushed into the
-        # autoscaling_policy. Need to discuss what the right API would look like.
-        upper_bound = self.get_capacity_adjusted_num_replicas(
-            autoscaling_policy.config.max_replicas, target_capacity
-        )
-        if (
-            target_capacity_scale_direction == TargetCapacityScaleDirection.UP
-            and autoscaling_policy.config.initial_replicas is not None
-        ):
-            lower_bound = self.get_capacity_adjusted_num_replicas(
-                autoscaling_policy.config.initial_replicas, target_capacity
-            )
-        else:
-            lower_bound = self.get_capacity_adjusted_num_replicas(
-                autoscaling_policy.config.min_replicas, target_capacity
-            )
-
-        clipped_decision_num_replicas = max(
-            lower_bound, min(decision_num_replicas, upper_bound)
-        )
-
-        if (
-            clipped_decision_num_replicas == self._target_state.num_replicas
-            and self._target_state.adjust_capacity is False
-        ):
+        if decision_num_replicas == self._target_state.target_num_replicas:
             return
 
         logger.info(
             f"Autoscaling replicas for deployment {self.deployment_name} in "
-            f"application {self.app_name} to {clipped_decision_num_replicas}. "
+            f"application {self.app_name} to {decision_num_replicas}. "
             f"current_num_ongoing_requests: {current_num_ongoing_requests}, "
             f"current handle queued queries: {current_handle_queued_queries}."
         )
 
         new_info = copy(self._target_state.info)
-        new_info.set_autoscaled_num_replicas(clipped_decision_num_replicas)
         new_info.version = self._target_state.version.code_version
+
+        has_reached_autoscaling_bounds = self._has_reached_autoscaling_bounds()
+        if has_reached_autoscaling_bounds:
+            new_status_trigger = DeploymentStatusTrigger.AUTOSCALING
+        else:
+            new_status_trigger = self.curr_status_info.status_trigger
```

Comment:
Nit: this isn't strictly necessary since if `allow_scaling_statuses = False`, `set_target_state` won't touch the status.